### DISCUSSION TITLE
IconLayer not working with MVTLayer

### DISCUSSION THREAD

**Discussion by samuel-litzler** [visgl/deck.gl#7247] [Q&A]
Hi all,
(french dev)
I use deckGL in my company especially with MVT layers.

I need to display MVT Layer on **GoogleMaps** with Icons : 
![image](https://user-images.githubusercontent.com/24872653/188850772-53c3ecbe-d2c3-46b3-9d65-6360c43ff06a.png)

I download the min.js of the lastest version because i'm using JS without a framework.
SO i need to instanciate Layer like this : 
![image](https://user-images.githubusercontent.com/24872653/188851218-1ae89daf-23ba-4dc3-a044-a8cc441d494a.png)

I see in the doc that to display ICONS with MVT layer i need to user GeoJsonLayer with `pointType: 'icon'`

Note : My data url is build dynamicaly and i use a function to declare initialProps for my MVTLayer.

So here is my code : 
![image](https://user-images.githubusercontent.com/24872653/188851878-7ce7c0b1-b70c-4fa3-a503-f12c11d946e8.png)

Sadly that doesnt works with icon... (with point and circle its work)
Here is the error message : 

![image](https://user-images.githubusercontent.com/24872653/188852356-1fbbefd9-a2ed-494c-9f95-e91525b109f8.png)

So i think because i'm using JS without a framework and to instanciate a Layer i need to do : `deck.[LayerName]`, so the IconLayer is never instanciate.

Any solutions to this ?

**Comment by Pessimistress** [comment]
You do not need to use `_subLayerProps`. See https://deck.gl/docs/api-reference/layers/geojson-layer#pointtypeicon-options
